### PR TITLE
Add slim images

### PR DESCRIPTION
**Pull request by ianpartridge**
Add "slim" images for Ubuntu 16.04 and 18.04.  Slim images are images designed just for running an already built Swift program. Consequently, they do not contain the Swift compiler.

The normal and slim images can be combined via a multi-stage `Dockerfile` to produce a lighter-weight image ready for deployment. For example:

```dockerfile
FROM swift:latest as builder
WORKDIR /root
COPY . .
RUN swift build

FROM swift:slim
WORKDIR /root
COPY --from=builder /root .
CMD [".build/x86_64-unknown-linux/debug/docker-test"]
```

Prior discussion: https://forums.swift.org/t/idea-runtime-slim-images/22695

### CODE REVIEW

**Comment by symbiont-haris-amin** [comment]
@ianpartridge this is fantastic! What are the slimmed down image sizes now?

**Comment by hamin** [comment]
@ianpartridge this is great! What are the slimmed down image sizes now?

**Comment by wlisac** [comment]
@ianpartridge this is really great! Very excited to give this a try 👌

**Comment by ianpartridge** [comment]
16.04 slim: 249MB
18.04 slim: 226MB

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

File: 5.0/ubuntu/16.04/Dockerfile-slim [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    apt-get -q install -y \
+    libatomic1 \
+    libbsd0 \
+    libcurl3 \
+    libxml2 \
+    tzdata \
+    && rm -r /var/lib/apt/lists/*
+
+# Everything up to here should cache nicely between Swift versions, assuming dev dependencies change little
+ARG SWIFT_PLATFORM=ubuntu18.04
```
Comment (ianpartridge) [inline 5.0/ubuntu/16.04/Dockerfile-slim]: Oops this is wrong.

**Comment by ianpartridge** [comment]
@swift-ci test

**Review by shahmishal — APPROVED** [review]
Awesome! Thanks!

**Comment by shahmishal** [comment]
Can you add information about the slim docker image in the readme?

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by satishbabariya** [comment]
@ianpartridge  isn't it better to have Dockerfile in the slim directory rather than having **Dockerfile-slim**, i.e

swift-docker/5.0/ubuntu/bionic/Dockerfile
swift-docker/5.0/ubuntu/bionic-slim/Dockerfile
swift-docker/5.0/alpine/Dockerfile

or in current way:
swift-docker/5.0/ubuntu/18.04/Dockerfile
swift-docker/5.0/ubuntu/18.04/slim/Dockerfile

**Comment by ianpartridge** [comment]
Yes it looks like that's how Python and Ruby do it.  I'll push an update to reorganise things.

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Comment by ianpartridge** [comment]
@swift-ci test

**Pull request merged**